app/src/main/kotlin/com/softteco/template/MainActivity.kt

```diff
@@ -49,7 +49,7 @@ class MainActivity : ComponentActivity() {
                 }
                 isUserLoggedIn?.let {
                     val startDestination = if (it) Graph.BottomBar.route else Graph.Login.route
-                    AppContent(startDestination)
+                    AppContent(startDestination, { isUserLoggedIn = it })
```

Comment:
Why changing `isUserLoggedIn` via callback works better then 
`isUserLoggedIn = profileRepository.getUser() is Result.Success`?
It seems they should work similarly.

Reply:
It's a good catch! 
Could we add some SessionManager to handle if a user logged in? The SessionManager could be a singleton, injected to MainActivity and in places where we do login\logout. In MainActivity we will just collect a state field of SessionManager. 
So in this way, we will avoid adding the callback for handling whether the user logged in or logged out